src/Components/Web.JS/src/Boot.WebAssembly.ts

```diff
@@ -119,7 +119,7 @@ async function boot(options?: Partial<WebAssemblyStartOptions>): Promise<void> {
   }
 
   // Start up the application
-  platform.callEntryPoint(resourceLoader.bootConfig.entryAssembly);
+  await platform.callEntryPoint(resourceLoader.bootConfig.entryAssembly);
```

Comment:
@TanayParikh @pranavkm Does this change the semantics of app startup? It depends on how `BINDING.call_assembly_entry_point`, which I can't guess from reading the diff.

I thought that, previously, calling `Blazor.start` would return a promise that resolves when the .NET code begins running. Is that still how it behaves, or does it now resolve only when `MainAsync`'s task completes?

Reply:
This seems to be related to the functionality of `BINDING.call_assembly_entry_point` specifically. Pinging @lewing for more context on this given https://github.com/dotnet/runtime/pull/44045.

Reply:
@SteveSandersonMS is correct, successfully awaiting that task should give you the exit code, control will return as soon as it can, given that the runtime is currently singly threaded.  If your model expects control to yield here before the runtime has exited, don't await the entry point promise.  That said, any guarantees that the entry point will yield at a specific point is probably weak.

Reply:
Thanks @lewing 

In this case, this PR is a breaking change we should fix before preview 4 ships. @TanayParikh I'll follow up by email to coordinate.

Reply:
https://github.com/dotnet/aspnetcore/issues/31971